File: app/src/test/java/net/tlalka/android/fiszki/models/entities/WordTest.java
package net.tlalka.android.fiszki.models.entities;

import net.tlalka.android.fiszki.models.types.LanguageType;
import net.tlalka.android.fiszki.models.types.OwnerType;
import org.junit.Test;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNotNull;

public class WordTest {

    @Test
    public void testCheekDefaultConstructor() {

        // when
        Word word = new Word();

        // then
        assertNotNull(word);
        assertEquals(0, word.getId());
        assertEquals(0, word.getProgress());
    }

    @Test
    public void testCheekCustomConstructor() {

        // given
        String value = "red";
        Cluster cluster = new Cluster();
        Lesson lesson = new Lesson();
        LanguageType languageType = LanguageType.EN;

        // when
        Word word = new Word(value, cluster, lesson, languageType);

        // then
        assertNotNull(word);
        assertEquals(value, word.getValue());
        assertEquals(cluster, word.getCluster());
        assertEquals(languageType, word.getLanguageType());
        assertEquals(OwnerType.SYSTEM, word.getOwnerType());
    }
}

File: app/src/test/java/net/tlalka/android/fiszki/model/entities/WordTest.kt
package net.tlalka.android.fiszki.model.entities

import net.tlalka.android.fiszki.model.types.LanguageType
import net.tlalka.android.fiszki.model.types.OwnerType
import org.junit.Assert.assertEquals
import org.junit.Assert.assertNotNull
import org.junit.Test

class WordTest {

    @Test
    fun testCheekDefaultConstructor() {

        // when
        val word = Word()

        // then
        assertNotNull(word)
        assertEquals(0, word.id)
        assertEquals(0, word.progress)
    }

    @Test
    fun testCheekCustomConstructor() {

        // given
        val value = "red"
        val cluster = Cluster()
        val lesson = Lesson()
        val languageType = LanguageType.EN

        // when
        val word = Word(value, cluster, lesson, languageType)

        // then
        assertNotNull(word)
        assertEquals(value, word.value)
        assertEquals(cluster, word.cluster)
        assertEquals(languageType, word.languageType)
        assertEquals(OwnerType.SYSTEM, word.ownerType)
    }
}
